javascripts/discourse/initializers/dice-setup.js

```diff
@@ -0,0 +1,178 @@
+import I18n from "I18n";
+import { getRegister } from "discourse-common/lib/get-owner";
+import { withPluginApi } from "discourse/lib/plugin-api";
+import WidgetGlue from "discourse/widgets/glue";
+
+import MersenneTwister from "discourse/lib/mersenne-twister";
+import murmurhash3 from "discourse/lib/murmurhash3";
+
+const MURMUR_HASH_SEED = 843031067;
+
+// [NdN] - base format, \1 is quantity \2 is faces
+// +N or -N - flat modifier, \3 is sign \4 is modifier
+// tN - report pass/fail, \5 is threshold
+// i - report individual dice, \6 is i
+// TODO - kN - keep top N results, \7 is keep count
+const diceRegexp = /(\d+)?d(\d+)?(?:([+-])(\d+))?(?:t(\d+))?(i)?/,
+  idxQuantity = 1,
+  idxFaces = 2,
+  idxModSgn = 3, idxModValue = 4,
+  idxThreshold = 5,
+  idxIndividual = 6;
+
+// returns Object{ quantity, faces, modValue, threshold, individual: Boolean, errors: []i18nkey }
+function parseDice(match) {
+  const errors = [];
+  if (!match) {
+    errors.push("dice.invalid.generic");
+    return { errors };
+  }
+  const parse = s => {
```

Comment:
I would recommend hoisting this function out, since it's not closing over any variables. The JIT probably doesn't care but why not make it's life easier (and the function shorter.)